src/workerd/jsg/util.c++

```diff
@@ -736,4 +736,84 @@ void returnRejectedPromise(
   return returnRejectedPromiseImpl(info, exception, tryCatch);
 }
 
+// ======================================================================================
+
+// The implementation here is modeled very closely after the implementation of the same
+// class in Node.js, with modifications to fit our conventions.
```

Comment:
TODO(now): Before this lands, the LICENSE file should be updated to reflect that some code from Node.js is used.